### PR TITLE
arch-riscv: Fix the RiscvBareMetal parameter reset_vect

### PR DESCRIPTION
**Pull request by rogerchang23424**
The `reset_vect` has exist for a long time and `reset_vect` will not effect if the user gonna to use customized reset_vect. The CL added the `auto_reset_vect` to let the config determine the `reset_vect` from workload entry point or user-specified

Ref: https://gem5-review.googlesource.com/c/public/gem5/+/42053

Change-Id: I928c0dc42aaa85ceabf8d75f9654486496e0ffee

### CODE REVIEW

File: src/arch/riscv/bare_metal/fs_workload.hh [hunk 1]
```diff
@@ -45,9 +45,9 @@ class BareMetal : public Workload
     // checker for bare metal application
     bool _isBareMetal;
     // entry point for simulation
-    Addr _resetVect;
     loader::ObjectFile *bootloader;
     loader::SymbolTable bootloaderSymtab;
+    Addr _resetVect;
```
Comment (powerjg) [inline src/arch/riscv/bare_metal/fs_workload.hh]: Is this change necessary? I guess it doesn't matter...

File: src/arch/riscv/RiscvFsWorkload.py [hunk 2]
```diff
@@ -43,6 +43,9 @@ class RiscvBareMetal(Workload):
     bootloader = Param.String("File, that contains the bootloader code")
     bare_metal = Param.Bool(True, "Using Bare Metal Application?")
     reset_vect = Param.Addr(0x0, "Reset vector")
+    auto_reset_vect = Param.Bool(
+        True, "Use bootloader entry point as reset vector"
```
Comment (powerjg) [inline src/arch/riscv/RiscvFsWorkload.py]: Could you add here "If this is true, the `reset_vect` parameter is ignored"

**Review by powerjg — COMMENTED** [review]
Two nits

**Review by hnpl — APPROVED** [review]

File: src/arch/riscv/RiscvFsWorkload.py (hunk 2 above)
Comment (rogerchang23424) [reply src/arch/riscv/RiscvFsWorkload.py]: Sure

**Review by hnpl — APPROVED** [review]

**Review by powerjg — APPROVED** [review]
